compiler/damlc/tests/src/DA/Test/DamlcMultiPackage.hs

```diff
@@ -0,0 +1,760 @@
+        "Multi project"
+        [ test "Build package B" ["--multi-package-search"] "./packages/package-b" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1"
+            , PackageIdentifier "lib-b" "0.0.1"
+            , PackageIdentifier "package-a" "0.0.1"
+            , PackageIdentifier "package-b" "0.0.1"
+            ]
+        , test "Build package A" ["--multi-package-search"] "./packages/package-a" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1"
+            , PackageIdentifier "lib-b" "0.0.1"
+            , PackageIdentifier "package-a" "0.0.1"
+            ]
+        , test "Build lib B" ["--multi-package-search"] "./libs/lib-b" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1"
+            , PackageIdentifier "lib-b" "0.0.1"
+            ]
+        , test "Build lib A" ["--multi-package-search"] "./libs/lib-a" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1" ]
+        , test "Build all from packages" ["--all"] "./packages" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1"
+            , PackageIdentifier "lib-b" "0.0.1"
+            , PackageIdentifier "package-a" "0.0.1"
+            , PackageIdentifier "package-b" "0.0.1"
+            ]
+        , test "Build all from libs" ["--all"] "./libs" multiProject $ Right
+            [ PackageIdentifier "lib-a" "0.0.1"
+            , PackageIdentifier "lib-b" "0.0.1"
+            ]
+        ]
+    , testGroup
+        "Cycle detection"
+        [ test "Multi-package project cycle from lib-a" ["--multi-package-search"] "./libs/lib-a" cyclicMultiPackage $ Left "Cycle detected"
+        , test "Multi-package project cycle from package-a" ["--multi-package-search"] "./packages/package-a" cyclicMultiPackage $ Left "Cycle detected"
+        , test "Multi-package project cycle from libs --all" ["--all"] "./libs" cyclicMultiPackage $ Left "Cycle detected"
+        , test "Multi-package project cycle from packages --all" ["--all"] "./packages" cyclicMultiPackage $ Left "Cycle detected"
+        , test "Package dep cycle from package-a" ["--multi-package-search"] "./package-a" cyclicPackagesProject $ Left "recursion detected"
+        , test "Package dep cycle from package-b" ["--multi-package-search"] "./package-b" cyclicPackagesProject $ Left "recursion detected"
+        , test "Package dep cycle from root --all" ["--all"] "" cyclicPackagesProject $ Left "recursion detected"
+        ]
+    , testGroup
+        "Special flag behaviour"
+        [ test "Multi-package build rebuilds dar with --output" ["--multi-package-search"] "./package-b" customOutPathProject
+            $ Right [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"]
+        , test "Build --all doesn't forward options flags like --ghc-options" ["--all", "--ghc-option=-Werror"] "" warningProject
+            $ Right [PackageIdentifier "package-a" "0.0.1", PackageIdentifier "package-b" "0.0.1"]
```

Comment:
just wondering, why shouldn't we forward these options?

Reply:
Fair point, this was an early assumption that I didn't question.
It might be possible to forward to all invocations, but it'd be difficult, since we'd need to use the `optionsParser` to identify which string flags are acceptable, then forward the original flags filtered through to the sub-process calls.
Also note that given backwards compat with sdk versions, this would break if we ever added more flags to `optionsParser`.
I'd prefer to disallow specifying options flags when using --all, but that's not written here.

Reply:
> It might be possible to forward to all invocations, but it'd be difficult, since we'd need to use the `optionsParser` to identify which string flags are acceptable, then forward the original flags filtered through to the sub-process calls

that's fair, it does sound complicated. In the next two cases the flags do get forwarded to `daml build` for the current directory, does that mean we just pass them on without parsing them? (not implying that's right/wrong, just wondering)

> Also note that given backwards compat with sdk versions, this would break if we ever added more flags to `optionsParser`.

oof, true

> I'd prefer to disallow specifying options flags when using --all, but that's not written here.

yes that would be nice if we go forward with the current approach

Reply:
> does that mean we just pass them on without parsing them?
No. For the "main" package when we're running without `--all`, we don't call to subprocess, as daml-assistant will already ensure we're running the correct sdk version before we parse damlc flags. As such, we pass the already generated/parsed `Options` type directly to the build.

Reply:
related to this, I think it would be nice to let users set global flags in `multi-package.yaml`, which isn't that different.

> Also note that given backwards compat with sdk versions, this would break if we ever added more flags to `optionsParser`.

now I'm thinking, maybe it would be fine to forward every flag (minus the ones for `multi-package` itself) as-is and let each sdk's `daml build` figure it out? 

In any case, very much not part of this PR :)

Reply:
But thanks for bringing this up - I do now see there is value in having it, if it can be done